src/block/Sugarcane.php

```diff
@@ -123,4 +129,21 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 
 		return false;
 	}
+
+	private function hasNearbyWater(Block $down) : bool{
+		foreach($down->getHorizontalSides() as $sideBlock){
+			$blockId = $sideBlock->getTypeId();
+			if($blockId === BlockTypeIds::WATER || $blockId === BlockTypeIds::FROSTED_ICE){
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public function onNearbyBlockChange() : void {
```

Comment:
```suggestion
	public function onNearbyBlockChange() : void{
```

Also formatting issue.